app/code/Magento/CustomerImportExport/Model/Import/Customer.php

```diff
@@ -569,11 +569,11 @@ public function getEntityTable()
      */
     public function getValidColumnNames()
     {
-        $this->validColumnNames = array_merge(
-            $this->validColumnNames,
-            $this->customerFields
+        return array_unique(
+            array_merge(
+                $this->validColumnNames,
+                $this->customerFields
```

Comment:
Sorry, this wikipedia link is too complex for me. I understand what you mean about flipping the arrays themselves but that will make the code difficult to understand. It will also imply to make changes in several places where all this arrays are created/updated.
My goal was just to point out that saving recursively `$this->validColumnNames` including itself, creates an array that is twice bigger in every iteration.
How to fix that? there are several valid solutions including the 2 we are discussing. I just went for the simplest and most understandable but of course other are also valid.